### PR TITLE
176 bidding pallet intergrate share distributor pallet

### PR DESCRIPTION
**Pull request by letodunc**
Call share distributor from bidding pallet
Added complementary functions in housing fund to manage unreservation fund and purchase operation
Updated `Onboarding::do_buy()` with housing fund to manage fund unreserved for the purchase in `Housing Fund`
Updated `Share_Distributor::create_virtual()` with housing fund purchase process to keep transaction history and stay up to date
Updated tests to cover the updates

### CODE REVIEW

**Review by ndkazu — COMMENTED** [review]
I don't see any modification of `Share_Distributor::create_virtual()` as stated in the PR manifest: are you still working on it?

**Comment by letodunc** [comment]
> I don't see any modification of `Share_Distributor::create_virtual()` as stated in the PR manifest: are you still working on it?

No, but the change wasn't onboarded i the commit. Now you should see it.

File: pallets/housing_fund/src/lib.rs [hunk 1]
```diff
@@ -118,6 +118,8 @@ pub mod pallet {
 		/// Fund reservation succeded
 		FundReservationSucceeded(T::NftCollectionId, T::NftItemId, BalanceOf<T>, BlockNumberOf<T>),
 		FundReservationCancelled(T::NftCollectionId, T::NftItemId, BalanceOf<T>, BlockNumberOf<T>),
+		PurchaseFundValidated(T::NftCollectionId, T::NftItemId, BalanceOf<T>, BlockNumberOf<T>),
+		FundUnreservedForPurchase(T::NftCollectionId, T::NftItemId, BalanceOf<T>, BlockNumberOf<T>),
```
Comment (ilhanu) [inline pallets/housing_fund/src/lib.rs:122]: good addition

**Review by ilhanu — COMMENTED** [review]
It looks good, small changes requested

**Review by ndkazu — APPROVED** [review]

**Pull request merged**